design/concise-tidy-design.Rmd

```diff
@@ -29,13 +29,26 @@ The general principles I propose for tidying `crmPack` objects are as follows:
 * If the slot is scalar, the slot is converted to a 1x1 `tibble`.  This will ease downstream operations such as `row-bind`ing.
 * If the object being tidied contains multiple slots of (potentially) different lengths, the results is a list of `tibble`s.  The list may be nested to multiple levels.  (See, for example, `LogisticLogNormal`.)
 * The column names of the tidied `tibble` correspond to the slot names of the parent object.
+  * Exception: where the slot has a plural contains a `vector` or `list`, the column name will be singular.  See, for example, `CohortSizeParts` below.
 * If an object contains a vector or matrix of values that correspond to an "obvious" set of parameters, columns take on the names of those parameters and the `tibble` is augmented by a column named `Parameter`.  This most often happens in the mean vector and covariance matrix of `ModelParamsNormal`.
 * When the value of a slot has not been set, a zero-row `tibble` is returned.
 * When the value of a slot has scalar attributes, these attributes are added as a column of the `tibble`, whose name is the name of the attribute and whose value is the value of the attribute for every row of the tibble.  Vector attributes will have to be handled on a case-by-case basis.
 * `tbl_<className>` is prepended to the class of the tidy `tibble`.
 
 ### Exceptions
 *  Where a vector slot (or series of vector slots) define a range, then the naming convention described above is not followed.  Instead, columns named `min` and `max` define the extent of the range and a column named `Range` labels the range.
+  * Would, say, `left` and `right` be better names to avoid potential name clashes with the built-in `min` and `max` functions? 
+  
+Unfortunately, `crmPack` classes do not inherit from a common super class (which need not do anything other than identify the class as being part of the package).  Consequently, a single `crpPack`-specific default `tidy` method is not possible.  A separate, class-specific `tidy` method will be needed for each of the main `crmPack` super classes (`Data`, `GeneralModel`, `Increments`, `CohortSize`, `MaxDose`, `Stopping`, `Design`, `Samples`, `Simulations`, etc).  However, the basic functionality is the same regardless of super class: iterate over slots that are not functions, converting each to a tibble.  This common functionality could be propvided by a helper function.  See below. 
```

Comment:
Ah I have an idea for this. To avoid copy/paste application of helper function in top level class specific methods, we could define a [class union](https://stat.ethz.ch/R-manual/R-devel/library/methods/html/setClassUnion.html) of these super classes and just have a single method for it.

Reply:
That's neat!  I wasn't aware of class unions, but that would certainly simplify things.  I will adjust the text accordingly and open a separate issue for definition of the class union.

I'm not sure why you still see the HTML file.  I thought I'd sorted that with a new `.gitignore`....

Reply:
yeah you can always double check in the "Files changed" tab: 
<img width="970" alt="image" src="https://github.com/Roche/crmPack/assets/1071227/295a7d66-13d9-4d0a-95fe-8277407d9611">